### ISSUE TITLE
[apex] ApexDoc false-positive for the first method of an annotated Apex class

### ISSUE THREAD

**Issue by liqingjun123** [pmd/pmd#4774]
**Affects PMD Version:** 7.0.0-rc4

**Rule:** [ApexDoc](https://pmd.github.io/latest/pmd_rules_apex_documentation.html#apexdoc)

**Description:**

**Code Sample demonstrating the issue:**

```apex
/**
 * @description       : rest service
 **/
@RestResource(urlMapping='/api/v1/get/*') 
global without sharing class TestService {
    /**
     * @description Bar
     */
    public void bar() { return; }
}
``` 

**Expected outcome:**

PMD reports a violation at line 9, but that's wrong. That's a false positive.

```
"rest.cls","3","9","Missing ApexDoc comment","Documentation","ApexDoc"
```

**Running PMD through:** *[CLI]*

**Comment by liqingjun123** [comment]
`/**
 * @description       : rest service
 **/
@RestResource(urlMapping='/api/v1/get/*') 
global without sharing class TestService {
    /**
     * @description Bar
     * @return Bar
     */
    public void bar() { return; }
}`

**Comment by liqingjun123** [comment]
Hi Team,  @clement.fournier@tu-dresden.de could you please help on the issue ASAP? Thanks very much

**Comment by jsotuyod** [comment]
This is not a false positive. As per the rule's documentation, the rule will flag:

> * ApexDoc comments on void or constructor methods should not contain @return.

Your sample `bar` method's doc is effectively a violation.

**Comment by liqingjun123** [comment]
> This is not a false positive. As per the rule's documentation, the rule will flag:
> 
> > * ApexDoc comments on void or constructor methods should not contain @return.
> 
> Your sample `bar` method's doc is effectively a violation.

I have updated the comments for the function bar method. Now the issue still exists.

**Comment by adangel** [comment]
I can confirm: If the annotation `@RestResource` is removed, then the formal parameter of the method is present in the AST and the rule works. But if the annotation is added, then the formal parameter of the method is missing in the AST...

I guess, there is a bug when we map the comments to the (nearest) node.

**Comment by adangel** [comment]
Thanks for the reminder. I've just verified, that this is actually already fixed in the current master branch, probably with the switch from Jorje to apex-parser (#3766).
